### PR TITLE
[BUGFIX] Ensure that result_format from saved expectation suite json file takes effect

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-330
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/validator/validator.py
Comment (donaldheppner) [inline great_expectations/validator/validator.py:99]: Can this try/except be replaced with an if statement? (if evaluated_expectations != 0)

Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:99]: @donaldheppner Potentially -- I am not sure right now as it would need to go into a separate effort/PR (because I did not write this method -- I only moved it to the section of the module where all private methods are placed together -- for code beautification purposes).  Thanks!

Comment (donaldheppner) [reply great_expectations/validator/validator.py:99]: okay

File: great_expectations/validator/validator.py
Comment (talagluck) [inline great_expectations/validator/validator.py:1697]: Should we include `include_config` here as well? Do we need `meta` here too?

File: great_expectations/validator/validator.py
Comment (talagluck) [inline great_expectations/validator/validator.py:1716]: This check seems odd to me. Why are we checking whether the `result_format` from `self.default_expectation_args` is equal to that of `Validator.DEFAULT_RUNTIME_CONFIGURATION`? Could this expression be made more concise?

Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:1716]: @talagluck This makes sure that things did not become "dirty".  It is important to figure out what changed.  This is a consequence of having default values that may not be what the user saved in the `JSON` file -- and the fact that there are overrides possible in `Validator.validate()`.

File: great_expectations/validator/validator.py
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:1697]: @talagluck If we do any functionality increase, then we need to do it in a separate PR.  Specifically to `include_config`, I would say "NO" -- because it does NOT impact validation and how the validation results are returned.  This is why it is not there to begin with, I believe.  Good question though (I thought about it too).  Thanks.

Comment (talagluck) [reply great_expectations/validator/validator.py:1697]: I think we may need `include_config`, and if we do not include `include_config` here, we should have a comment as to why, since `include_config` is the third of three `runtime_kwargs`. It may also be worth looking at whether `include_config` should be a top level `runtime_kwarg`, as it might make more sense as a component of the `result_format` dictionary.

File: tests/expectations/test_expectation_arguments.py
Comment (talagluck) [inline tests/expectations/test_expectation_arguments.py]: Please change all instances where `result_format` is string to dictionary.

File: tests/expectations/test_expectation_arguments.py
Comment (talagluck) [inline tests/expectations/test_expectation_arguments.py]: For the various `result_format` tests, please add a literal check for the result object - a comparison between the expected and actual ExpectationValidationResults.result object, so that it is clear from the test how each object is returned.

File: great_expectations/validator/validator.py
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:1697]: Comment added for discussion and architectural review.

File: tests/expectations/test_expectation_arguments.py
Comment (alexsherstinsky) [reply tests/expectations/test_expectation_arguments.py]: Done (in next commit).

File: tests/expectations/test_expectation_arguments.py
Comment (alexsherstinsky) [reply tests/expectations/test_expectation_arguments.py]: Done.

File: tests/expectations/test_expectation_arguments.py
Comment (talagluck) [inline tests/expectations/test_expectation_arguments.py:107]: Thanks!

File: tests/expectations/test_expectation_arguments.py
Comment (talagluck) [inline tests/expectations/test_expectation_arguments.py:543]: Awesome!

File: tests/expectations/test_expectation_arguments.py
Comment (talagluck) [inline tests/expectations/test_expectation_arguments.py:645]: For this one, the assertion as an empty dict looks a little strange. Could you demonstate that `result` has the expected boolean?

**Review by talagluck — COMMENTED** [review]
Almost there! Just one more request to slightly modify tests.

File: tests/expectations/test_expectation_arguments.py
Comment (alexsherstinsky) [reply tests/expectations/test_expectation_arguments.py:645]: @talagluck The empty `dict` in `result.result` is correct for `BOOLEAN_ONLY`.  Just added the assertion on the `boolean` flag (committed).  Please note that, importantly, the `result` object contains the auxiliary information (`ExpectationValidationResult`), while the `result.result` is a dictionary containing the detailed information about the validation.  In case `result_format` is `COMPLETE`, this dictionary is long; in case `result_format` is `SUMMARY` or `BASIC`, this dictionary is smaller, and in case `result_format` is `BOOLEAN_ONLY`, this dictionary is empty.  Thanks.

**Review by anthonyburdi — APPROVED** [review]
LGTM - pending re-review by @talagluck

File: tests/expectations/test_expectation_arguments.py
Comment (talagluck) [reply tests/expectations/test_expectation_arguments.py:645]: Thanks, Alex! One last request - can we have a test asserting that `result` is equal to its expected value? It would be great to have a clear example in our tests of what we expect a `BOOLEAN_ONLY` result to look like, in case we need to point users to this. This will ensure that your explanation here is clear from the test case.

Comment (alexsherstinsky) [reply tests/expectations/test_expectation_arguments.py:645]: @talagluck Great idea (for everybody) -- adding (in next commit).

Comment (alexsherstinsky) [reply tests/expectations/test_expectation_arguments.py:645]: @talagluck Changes committed.  Thanks.

**Review by talagluck — APPROVED** [review]
LGTM! Thanks for the updates!

**Pull request merged**
